tests/test0185.d

```diff
@@ -0,0 +1,27 @@
+//T compiles:yes
+//T retval:0
+//T has-passed:yes
+// Tests nested switch 
+int main() {
+	int x = 10;
+	switch (x) {
+   		case 1 :
+      			switch (x) { 
+      				default : 
+		        	      break; 
+      			}
+			break;
+     		case 2 : 
+			switch (x) {
+				default : 
+					break;
+     			}
+			break;
+
+		default : 
+			break;
+  	}
+	
+	return 0;
+}
+
```

Comment:
Please add a test case where there is a default in a switch BEFORE a nested switch, in order to test that the flag is restored properly.